src/deferred-binding/storage-blob/azureStorageBlobClient.ts

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License.
+
+import { BlobClient, ContainerClient, StoragePipelineOptions } from '@azure/storage-blob';
+import { StorageBlobServiceClientStrategy } from './storageBlobServiceClientStrategy';
+
+export class AzureStorageBlobClient {
+    private blobClient: BlobClient | undefined;
+    private containerClient: ContainerClient | undefined;
+
+    /**
+     * Creates a new AzureStorageBlobClient instance
+     *
+     * @param strategyOrAccountUrl - The strategy to use for creating the BlobServiceClient or the account URL
+     * @param credentialOrOptions - The credential to use for authentication or storage pipeline options
+     * @param options - Storage pipeline options (optional, only used when the first parameter is an account URL)
+     */
+    constructor(
+        strategy: StorageBlobServiceClientStrategy,
+        containerName?: string,
```

Comment:
This allows for null values to be passed. This code wont return any client if any of these values are null. Either add a null check or we could remove the '?'.

Reply:
Yeah, good point. I will change in extensions repo (as code is moved there)